src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.service.impl.OperationStrategy;
+import core.basesyntax.service.impl.ShopService;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final Map<String, Integer> storage = new HashMap<>();
```

Comment:
Encapsulation principle issue: The `storage` map should be private and not be directly exposed. Consider providing methods to interact with the storage without giving direct access to the underlying data structure. This will help encapsulate the storage logic within the service and prevent unwanted modifications from outside the class.